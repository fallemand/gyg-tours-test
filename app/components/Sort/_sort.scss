@import '../../styles/variables';
@import './sort__small';

@media ( min-width: $breakpoint-desktop ) {
  @import './sort__medium';
}

