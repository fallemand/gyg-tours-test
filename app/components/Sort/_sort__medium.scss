@import '../../styles/variables';

.sort {
  width: 200px;
  display: inline-block;
  margin-right: 15px;

  .sort__field {
    margin: 5px 0;
    border-radius: 4px 0 0 4px;
    -webkit-appearance: none;
  }

  .sort__order {
    margin: 5px 0 5px -1px;
    border-radius: 0 4px 4px 0;
  }
}
