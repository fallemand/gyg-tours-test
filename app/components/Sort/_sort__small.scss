@import '../../styles/variables';

.sort {
  width: 100%;

  .sort__field {
    width: 85%;
    display: inline-block;
  }

  .sort__order {
    border: 1px solid $border-color;
    border-radius: 2px;
    color: $font-color-primary;
    height: 30px;
    padding: 5px;
    cursor: pointer;
    vertical-align: bottom;
    width: 15%;
    margin: 10px 0 10px -1px;
    background-color: #fff;

    &:hover {
      background-color: $background-color-hover;
    }
  }

  .sort__order-img {
    width: 100%;
    height: auto;
    max-height: 20px;
    max-width: 25px;
  }
}
