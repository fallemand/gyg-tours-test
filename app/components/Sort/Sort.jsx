import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import Dropdown from '../Dropdown';
import './_sort.scss';
import sortOrderImg from '../../assets/images/up-down.png';

class Sort extends React.Component {
  constructor(props) {
    super(props);
    this.handleDropdownChange = this.handleDropdownChange.bind(this);
    this.handleSortOrderChange = this.handleSortOrderChange.bind(this);
  }

  handleDropdownChange(query) {
    const { onChange, order } = this.props;
    onChange({ field: query, order });
  }

  handleSortOrderChange() {
    const { onChange, field, order } = this.props;
    onChange({ field, order: !order });
  }

  render() {
    const { className, fields, field } = this.props;
    return (
      <div className={classnames(className, 'sort')}>
        <Dropdown
          name="sort"
          title="Sort"
          className="sort__field"
          value={field}
          options={fields}
          onChange={this.handleDropdownChange}
        />
        <button
          className="sort__order"
          title="Change sort direction"
          type="button"
          onClick={this.handleSortOrderChange}
        >
          <img className="sort__order-img" src={sortOrderImg} alt="Up - Down" />
        </button>
      </div>
    );
  }
}

Sort.propTypes = {
  className: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  field: PropTypes.string,
  order: PropTypes.bool,
  fields: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  })).isRequired,
};

Sort.defaultProps = {
  className: null,
  order: false,
  field: '',
};

export default Sort;
