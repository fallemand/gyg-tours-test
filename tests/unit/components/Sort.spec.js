import React from 'react';
import Sort from '../../../app/components/Sort';

describe('Sort', () => {
  const props = {
    onChange: jest.fn(),
    field: '__VAL1__',
    order: true,
    fields: [
      { value: '__VAL1__', label: '__LAB1__' },
      { value: '__VAL2__', label: '__LAB2__', selected: true },
      { value: '__VAL1__', label: '__LAB1__', disabled: true },
    ],
  };

  it('render component', () => {
    const component = shallow(<Sort {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('render with className', () => {
    const component = shallow(<Sort className="__CLASSNAME__" {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('call onChange when click sort order button', () => {
    const component = shallow(<Sort {...props} />);
    component.find('.sort__order').simulate('click');
    expect(props.onChange).toHaveBeenCalled();
  });
});
