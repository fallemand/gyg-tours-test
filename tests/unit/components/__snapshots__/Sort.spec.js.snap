// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort render component 1`] = `
"<div className='sort'>
  <Dropdown name='sort' title='Sort' className='sort__field' value='__VAL1__' options={{...}} onChange={[Function: bound handleDropdownChange]} />
  <button className='sort__order' title='Change sort direction' type='button' onClick={[Function: bound handleSortOrderChange]}>
    <img className='sort__order-img' src={{...}} alt='Up - Down' />
  </button>
</div>"
`;

exports[`Sort render with className 1`] = `
"<div className='__CLASSNAME__ sort'>
  <Dropdown name='sort' title='Sort' className='sort__field' value='__VAL1__' options={{...}} onChange={[Function: bound handleDropdownChange]} />
  <button className='sort__order' title='Change sort direction' type='button' onClick={[Function: bound handleSortOrderChange]}>
    <img className='sort__order-img' src={{...}} alt='Up - Down' />
  </button>
</div>"
`;
